spacer/extract_features_utils.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+
+def gray2rgb(im):
+    """
+    Convert gray image to RGB image
+    :param im: gray image to be converted
+    :return: RGB image
+    """
+    w, h = im.shape
+    ret = np.empty((w, h, 3), dtype=np.uint8)
+    ret[:, :, 0] = im
+    ret[:, :, 1] = im
+    ret[:, :, 2] = im
+
+    return ret
+
+
+def crop_patch(im, rowcols, crop_size, scale, offsets=None):
```

Comment:
since we only support `scale=1`, go ahead and remove this parameter. We can always add it in later if needed.

Reply:
done